### PR TITLE
Misra 15 7

### PR DESCRIPTION
**Pull request by rbiasini**
`An "if... else if" statement shall always be terminated with an "else".`

Avoided blank `else` statements.

### CODE REVIEW

File: board/safety/safety_gm.h [hunk 1]
```diff
@@ -69,7 +69,8 @@ static void gm_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
     // res/set - enable, cancel button - disable
     if ((buttons == 2) || (buttons == 3)) {
```
Comment (geohot) [inline board/safety/safety_gm.h]: switch buttons?

File: board/safety/safety_honda.h [hunk 2]
```diff
@@ -31,7 +31,8 @@ static void honda_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
     int buttons = (to_push->RDLR & 0xE0) >> 5;
     if ((buttons == 4) || (buttons == 3)) {
```
Comment (geohot) [inline board/safety/safety_honda.h]: switch buttons

File: board/safety/safety_toyota.h [hunk 3]
```diff
@@ -126,12 +125,13 @@ static int toyota_tx_hook(CAN_FIFOMailBox_TypeDef *to_send) {
...
-      if (controls_allowed && long_controls_allowed) {
-        bool violation = max_limit_check(desired_accel, TOYOTA_MAX_ACCEL, TOYOTA_MIN_ACCEL);
-        if (violation) {
+      if (!controls_allowed || !long_controls_allowed) {
+        if (desired_accel != 0) {
           tx = 0;
         }
-      } else if (desired_accel != 0) {
+      }
+      bool violation = max_limit_check(desired_accel, TOYOTA_MAX_ACCEL, TOYOTA_MIN_ACCEL);
```
Comment (geohot) [inline board/safety/safety_toyota.h]: I like this pattern of if statements at the highest scope possible.

File: board/safety/safety_gm.h (hunk 1 above)
Comment (rbiasini) [reply board/safety/safety_gm.h]: you mean changing the var name?

Comment (rbiasini) [reply board/safety/safety_gm.h]: oh, switch - case

**Pull request merged**